### PR TITLE
Update api.js 

### PR DESCRIPTION
**Pull request by WhiteClueGaming**
As many people abuse this api by adding users as bot. I have added a check for that and if user is not a bot it will return **404 bot not found**!
Now Dan, you can merge this pr and enable the api for adding bots stats again :)

### CODE REVIEW

**Comment by yuansheng1549** [comment]
Just use client.users.fetch

**Comment by WhiteClueGaming** [comment]
Yes I can but there is no client on that router file and because of this api I used. Dans bot will be fine

**Comment by yuansheng1549** [comment]
client is defined in global variable

**Comment by WhiteClueGaming** [comment]
global.client

**Comment by Darker-Ink** [comment]
This still doesn't fix a ton of problems this has lmao

**Comment by freddierick** [comment]
Mhm it needs a way for users to variety they own a given bot 

I couldn't figure out a way to do it without the use of a self bot

**Comment by WhiteClueGaming** [comment]
Theirs some api for that

**Comment by WhiteClueGaming** [comment]
Ah we can't without making a self bot

**Comment by freddierick** [comment]
The least invasive approach is telling the user to create a callback URL with a key in it (https://danbot.host/verifyBot/<key>), then using a self bot to call the api discord endpoint that will tell you if it's a valid endpoint or not.
If it's valid that means the user has to own the bot as they set the callback

**Comment by WhiteClueGaming** [comment]
Ok

File: routes/api.js [hunk 1]
```diff
@@ -29,6 +29,12 @@ Router.post("/bot/:ID/stats", /* rateLimit(10000, 2) , */ (req, res) => { // tem
             .send({error: true, message: "'bot_id' must be a snowflake"});
     }
 
+    fetch('https://api.discordz.xyz/discord/v1/user?id=' + ID).then(a => a.json()).then(async checkIsBot => {
+
+    if(!checkIsBot || !checkIsBot.data || !checkIsBot.data["bot"] || (checkIsBot.data["bot"] !== true)) return res
+            .status(404)
+            .send({error: true, message: "Bot not found"});
+
```
Comment (itzJOHv) [inline routes/api.js:37]:
```suggestion
    const isBot = await bot.users.fetch(ID).then((u) => u.bot).catch(() => false);
    if (!isBot) return res.status(404).send({error: true, message: "Bot not found"});
```

**Review by itzJOHv — CHANGES_REQUESTED** [review]

**Comment by itzJOHv** [comment]
> The least invasive approach is telling the user to create a callback URL with a key in it ([https://danbot.host/verifyBot/[some](https://danbot.host/verifyBot/%5Bsome) key>), then using a self bot to call the api discord endpoint that will tell you if it's a valid endpoint or not. If it's valid that means the user has to own the bot as they set the callback

update dbh package to send to send owner id from client.application?

**Pull request closed without merging**